### PR TITLE
bump golang mock to 1.4.0

### PR DESCRIPTION
**Pull request by wawa0210**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

 /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

https://github.com/golang/mock/releases/tag/v1.4.0

**Does this PR introduce a user-facing change?**:

```release-note
bump golng mock to 1.4.0
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
why are we updating this?
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md#justifications-for-an-update

- is golang mock a user facing change?
- also there's a consistent typo in `golang` as `golng`

**Comment by wawa0210** [comment]
According to the mock release note, 1.4 is a relatively large version change, there are many bug fixes and refactorings

**Comment by BenTheElder** [comment]
I see many listed, but do any of them actually affect us in a positive way?

**Comment by BenTheElder** [comment]
cc @liggitt (please triage to someone else if you know someone else familiar with our use of this library!)

File: go.mod [hunk 1]
```diff
@@ -572,6 +572,8 @@ replace (
 	mvdan.cc/lint => mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b
 	mvdan.cc/unparam => mvdan.cc/unparam v0.0.0-20190209190245-fbb59629db34
 	rsc.io/pdf => rsc.io/pdf v0.1.1
+	rsc.io/quote/v3 => rsc.io/quote/v3 v3.1.0
+	rsc.io/sampler => rsc.io/sampler v1.3.0
```
Comment (liggitt) [inline go.mod]: this adds two new dependencies. unless there's a compelling reason to pick up the new gomock version, I would tend not to

**Pull request closed without merging**